WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/ProductPriceSettingsViewModel.swift

```diff
@@ -0,0 +1,219 @@
+import Yosemite
+
+/// Provides data needed for price settings.
+///
+protocol ProductPriceSettingsViewModelOutput {
+    var regularPrice: String? { get }
+    var salePrice: String? { get }
+    var dateOnSaleStart: Date? { get }
+    var dateOnSaleEnd: Date? { get }
+    var taxStatus: ProductTaxStatus { get }
+    var taxClass: TaxClass? { get }
+}
+
+/// Handles actions related to the price settings data.
+///
+protocol ProductPriceSettingsActionHandler {
+    // Initialization
+    func retrieveProductTaxClass(completion: @escaping () -> Void)
+
+    // Input field actions
+    func handleRegularPriceChange(_ regularPrice: String?)
+    func handleSalePriceChange(_ salePrice: String?)
+    func handleTaxClassChange(_ taxClass: TaxClass?)
+    func handleTaxStatusChange(_ taxStatus: ProductTaxStatus)
+    func handleScheduleSaleChange(isEnabled: Bool)
+    func handleSaleStartDateChange(_ date: Date)
+    func handleSaleEndDateChange(_ date: Date)
+
+    // Navigation actions
+    func completeUpdating(onCompletion: ProductPriceSettingsViewController.Completion, onError: (ProductPriceSetingsError) -> Void)
+    func hasUnsavedChanges() -> Bool
+}
+
+/// Error cases that could occur in product price settings.
+///
+enum ProductPriceSetingsError: Error {
+    case salePriceWithoutRegularPrice
+    case salePriceHigherThanRegularPrice
+}
+
+/// Provides view data for price settings, and handles init/UI/navigation actions needed in product price settings.
+///
+final class ProductPriceSettingsViewModel: ProductPriceSettingsViewModelOutput {
```

Comment:
Thank you for refactoring this, @jaclync! I love that it has unit tests to go with it. 💜